department-of-veterans-affairs/caseflow
# Mark tasks assigned to user as in progress when viewing appeal

Resolves #14023

### Description
Also adds back the bolding of case details links that have not yet been viewed.

### Acceptance Criteria
- [ ] Ama tasks assigned to user get updated to "in_progress" when a user veiws an appeal
- [ ] Cases that have not yet been viewed show up as bolded in the users queue
    - [ ] AMA tasks
    - [ ] Legacy tasks
- [ ] Cases that have been viewed show up as not bolded in the users queue
    - [ ] AMA tasks
    - [ ] Legacy tasks

### Testing Plan
1. Sign in as an attorney (BVASCASPER1) and go to your queue
2. All cases should be bolded
3. Pick a bolded ama appeal and check the tasks and appeal views
```ruby
uuid = ""
appeal = Appeal.find_by_uuid(uuid)
appeal.tasks.where(assigned_to: User.first).pluck(:status)
=> ["assigned"]
appeal.appeal_views.find_by(user: User.first)
=> nil
```
4. Click into the ama appeal's case details
5. Ensure this task is updated to in progress and an appeal view is created
```ruby
appeal.reload.tasks.where(assigned_to: User.first).pluck(:status)
=> ["in_progress"]
appeal.appeal_views.find_by(user: User.first)
=> => #<AppealView:0x00007fc5e7b3a330 ... >
```
7. Go back to your queue and make sure the link is no longer bolded

1. Pick a bolded legacy appeal and check the appeal views
```ruby
vacols_id = ""
appeal = LegacyAppeal.find_by_vacols_id(vacols_id)
appeal.appeal_views.find_by(user: User.first)
=> nil
```
2. Click into an legacy appeal's case details
3. Check the appeal views again
```ruby
appeal.reload.appeal_views.find_by(user: User.first)
=> #<AppealView:0x00007fc5f0670e90 ... >
```
4. Go back to your queue and make sure the link is no longer bolded

1. Pick a bolded legacy appeal and check the appeal views
```ruby
vacols_id = ""
appeal = LegacyAppeal.find_by_vacols_id(vacols_id)
appeal.appeal_views.find_by(user: User.first)
=> nil
```
2. Click into an legacy appeal's documents
3. Check the appeal views again
```ruby
appeal.appeal_views.find_by(user: User.first)
=> #<AppealView:0x00007fc5eeae3628 ... >
```
4. Go back to your queue and make sure the link is no longer bolded

### User Facing Changes
 - [x] Unviewed appeals are bolded in queue
 - [x] Viewed appeals are not bolded in queue

![ezgif-3-9d944c5d31f3](https://user-images.githubusercontent.com/45575454/81850269-c0d90c80-9525-11ea-9ad2-10316cb99cf6.gif)
